WordPressVIPMinimum/Sniffs/Performance/WPQueryParamsSniff.php

```diff
@@ -8,52 +8,86 @@
 
 namespace WordPressVIPMinimum\Sniffs\Performance;
 
-use WordPressVIPMinimum\Sniffs\Sniff;
-use PHP_CodeSniffer\Util\Tokens;
+use WordPressCS\WordPress\AbstractArrayAssignmentRestrictionsSniff;
 
 /**
  * Flag suspicious WP_Query and get_posts params.
  *
  *  @package VIPCS\WordPressVIPMinimum
  */
-class WPQueryParamsSniff extends Sniff {
+class WPQueryParamsSniff extends AbstractArrayAssignmentRestrictionsSniff {
 
 	/**
-	 * Returns an array of tokens this test wants to listen for.
+	 * Groups of variables to restrict.
+	 * This should be overridden in extending classes.
+	 *
+	 * Example: groups => array(
+	 *  'groupname' => array(
+	 *      'type'     => 'error' | 'warning',
+	 *      'message'  => 'Dont use this one please!',
+	 *      'keys'     => array( 'key1', 'another_key' ),
+	 *      'callback' => array( 'class', 'method' ), // Optional.
+	 *  )
+	 * )
 	 *
 	 * @return array
 	 */
-	public function register() {
+	public function getGroups() {
 		return [
-			T_CONSTANT_ENCAPSED_STRING,
+			// WordPress.com: https://lobby.vip.wordpress.com/wordpress-com-documentation/uncached-functions/.
+			// VIP Go: https://wpvip.com/documentation/vip-go/uncached-functions/.
+			'SuppressFilters' => [
+				'type'     => 'error',
+				'message'  => 'Setting `suppress_filters` to `true` is prohibited.',
+				'keys'     => [
+					'suppress_filters',
+				],
+				'callback' => [ $this, 'value_is_true' ],
+			],
+			'PostNotIn' => [
+				'type'    => 'warning',
+				'message' => 'Using `post__not_in` should be done with caution, see https://wpvip.com/documentation/performance-improvements-by-removing-usage-of-post__not_in/ for more information.',
+				'keys'    => [
+					'post__not_in',
```

Comment:
Since `post__not_in` and `exclude` are the same, should we punt it into one group?

Reply:
I considered that, but we'd need to have a careful think about the warning message in that case as using the existing message would be confusing.

Reply:
We can definitely revamp the warning message to something more ambiguous.

Reply:
I'll have a think about it. Suggestions welcome.

Reply:
Maybe something like...
"Using exclusionary parameters should be done with caution, see https://wpvip.com/documentation/performance-improvements-by-removing-usage-of-post__not_in/ for more information."

We can even re-vamp the documentation a bit to account for it as well.